scalopus_tracing/src/trace_configurator.cpp

```diff
@@ -94,9 +102,12 @@ std::map<unsigned long, TraceConfigurator::AtomicBoolPtr> TraceConfigurator::get
   return thread_state_;
 }
 
-TraceConfigurator& TraceConfigurator::getInstance()
+TraceConfigurator::Ptr TraceConfigurator::getInstance()
 {
-  static TraceConfigurator instance;
+  // https://stackoverflow.com/questions/8147027/
+  // Trick to allow make_shared with a private constructor.
+  struct make_shared_enabler : public TraceConfigurator {};
```

Comment:
Maybe do something like this, where the templated struct code be defined somewhere else only once:
``` c++
template <class T>
struct make_shared_enabler : public T {};

static TraceConfigurator::Ptr instance{std::make_shared<make_shared_enabler<TraceConfigurator>>()};
```

Reply:
That doesn't work unfortunately :( 
```
error: call to implicitly-deleted default constructor of 'scalopus::make_shared_enabler<scalopus::TracePointCollectorNative>'
<snip>
<snip>scalopus/scalopus_tracing/src/native/tracepoint_collector_native.cpp:44:32: note: default constructor of 'make_shared_enabler<scalopus::TracePointCollectorNative>' is implicitly deleted because base class 'scalopus::TracePointCollectorNative' has an inaccessible default constructor
  struct make_shared_enabler : public T {};
                               ^
```

Leaving as is.